pkg/adapter/nimbus-kyverno/processor/kpbuilder.go

```diff
@@ -239,75 +351,322 @@ func cocoRuntimeAddition(np *v1alpha1.NimbusPolicy) ([]kyvernov1.Policy, error)
 	return kps, multierr.Combine(errs...)
 }
 
-func escapeToHost(np *v1alpha1.NimbusPolicy, rule v1alpha1.Rule) kyvernov1.Policy {
+func virtualPatch(np *v1alpha1.NimbusPolicy, logger logr.Logger) ([]kyvernov1.Policy, error) {
+	rule := np.Spec.NimbusRules[0].Rule
+	requiredCVES := rule.Params["cve_list"]
+	var kps []kyvernov1.Policy
+	resp, err := utils.FetchVirtualPatchData[[]map[string]any]()
+	if err != nil {
+		return kps, err
+	}
+	for _, currObj := range resp {
+		image := currObj["image"].(string)
+		cves := currObj["cves"].([]any)
+		for _, obj := range cves {
+			cveData := obj.(map[string]any)
+			cve := cveData["cve"].(string)
+			if utils.Contains(requiredCVES, cve) {
+				// create generate kyverno policies which will generate the native virtual patch policies based on the CVE's
+				karmorPolCount := 1
+				kyvPolCount := 1
+				netPolCount := 1
+				virtualPatch := cveData["virtual_patch"].([]any)
+				for _, policy := range virtualPatch {
+					pol := policy.(map[string]any)
+					policyData, ok := pol["karmor"].(map[string]any)
+					if ok {
+						karmorPol, err := generatePol("karmor", cve, image, np, policyData, karmorPolCount, logger)
+						if err != nil {
+							logger.V(2).Error(err, "Error while  generating karmor policy")
+						}
+						kps = append(kps, karmorPol)
+						karmorPolCount += 1
```

Comment:
Why do we need to prevent further execution? I think it's not good to stop the entire engine if we were unable to build, let's say any one kubearmor policy, we also have other policy in-place which are valid and should be deployed in the cluster, in case of a invalid policy we should show the logs to the user that there is a error in deploying this particular policy, and should continue processing other policies.